### PR TITLE
[3.13] gh-117482: Simplify the Fix For Builtin Types Slot Wrappers

### PR DESCRIPTION
**Pull request by ericsnowcurrently**
In gh-121602, I applied a fix to a builtin types initialization bug.  That fix made sense in the context of some broader future changes, but introduced a little bit of extra complexity.  For earlier versions those future changes are not relevant; we can avoid the extra complexity.  Thus we can revert that earlier change and replace it with one that is more focused and conceptually simpler.

That's this PR.  It's essentially the implementation of an idea that @markshannon pointed out to me.  Note that this change would be much smaller if we didn't have to deal with compatibility for builtin types that explicitly inherit tp slots (see `expect_manually_inherited()`).

* Issue: gh-117482

### CODE REVIEW

File: Objects/typeobject.c [hunk 1]
```diff
@@ -10877,6 +10862,63 @@ recurse_down_subclasses(PyTypeObject *type, PyObject *attr_name,
     return 0;
 }
 
+static int
+expect_manually_inherited(PyTypeObject *type, void **slot)
```
Comment (markshannon) [inline Objects/typeobject.c:10866]: This all seems a bit ad-hoc.
This is just for 3.13 though, and we will use something more robust for 3.14 onward?

**Review by markshannon — COMMENTED** [review]
Would it make sense to set a flag for each slot that is explicitly overridden when we first see the `PyTypeObject` struct. That way, we can refer to that during subsequent calls to know whether the field was set by the author or by the VM?

We only need about [80 bits per type](https://github.com/python/cpython/blob/main/Include/typeslots.h)

**Comment by markshannon** [comment]
Thinking about this so more, I think the problem is that we are attempting to initial static builtin types more than once.
We should only do it once, pre process. Then this problem goes away.

File: Objects/typeobject.c (hunk 1 above)
Comment (ericsnowcurrently) [reply Objects/typeobject.c:10866]: Correct.  For 3.14+ we do it differently.

**Comment by ericsnowcurrently** [comment]
> Thinking about this so more, I think the problem is that we are attempting to initial static builtin types more than once. We should only do it once, pre process. Then this problem goes away.

I agree, but this would probably be too much of a change for 3.13 and 3.12.

**Comment by markshannon** [comment]
Does this fix the issue?
If it does, could you add a test?

The change itself is a bit ad-hoc, but it looks fine.

**Comment by ericsnowcurrently** [comment]
> If it does, could you add a test?

The existing test does verify that the underlying is still fixed.  I'll add a test for the embedding case.

**Pull request merged**